src/System.Runtime.Extensions/src/System/AppDomain.cs

```diff
@@ -280,5 +285,39 @@ public void SetThreadPrincipal(IPrincipal principal)
                 _defaultPrincipal = principal;
             }
         }
+
+        public IPrincipal GetThreadPrincipal()
+        {
+            IPrincipal principal = _defaultPrincipal;
+            if (principal == null)
+            {
+                switch (_principalPolicy)
+                {
+                    case PrincipalPolicy.UnauthenticatedPrincipal:
+                        if (s_getUnauthenticatedPrincipal == null)
+                        {
+                            Type type = Type.GetType("System.Security.Principal.GenericPrincipal, System.Security.Claims", throwOnError: true);
+                            Volatile.Write(ref s_getUnauthenticatedPrincipal, (Func<IPrincipal>)Delegate.CreateDelegate(typeof(Func<IPrincipal>), type, "GetDefaultInstance"));
```

Comment:
Was there a particular reason that `ignoreCase: false` and `throwOnBindFailure: false` parameters were only passed to Delegate.CreateDelegate for `WindowsPrincipal` and not to `UnauthenticatedPrincipal`?

This is causes a test failure in uapaot: https://mc.dot.net/#/product/netcore/30/source/official~2Fdotnet~2Fcorefx~2Frefs~2Fheads~2Fmaster/type/test~2Ffunctional~2Filc~2F/build/20190205.1/workItem/System.Threading.Thread.Tests/analysis/xunit/System.Threading.Threads.Tests.ThreadTests~2FUnauthenticatedPrincipalTest

Reply:
I think it is just an oversight.

Reply:
https://github.com/dotnet/corefx/pull/35112/commits/f7ffc46b40ac4bd4e780169b90ea6937f6481480